lib/browser/api/web-contents.ts

```diff
@@ -792,8 +792,7 @@ WebContents.prototype._init = function () {
     }
   });
 
-  const event = process._linkedBinding('electron_browser_event').createEmpty();
-  app.emit('web-contents-created', event, this);
+  app.emit('web-contents-created', { preventDefault () {} }, this);
```

Comment:
Doesn't this require `sender`?

Reply:
if it does, then it's an existing bug that it doesn't have it.

Reply:
I think we can easily standardize in this PR though.  All events should have a `sender`

Reply:
i'm happy to change this particular event to have a `sender`, but standardizing all emitted events this way is beyond the scope of this PR.

* Some events that we emit do not have an `Event` object at all, ex: [process.on('loaded')](https://www.electronjs.org/docs/latest/api/process#event-loaded), [utilityProcess.on('exit')](https://www.electronjs.org/docs/latest/api/utility-process#event-exit), [webContents.on('did-create-window')](https://www.electronjs.org/docs/latest/api/web-contents#event-did-create-window), [autoUpdater.on('error')](https://www.electronjs.org/docs/latest/api/auto-updater#event-error), [app.on('session-created')](https://www.electronjs.org/docs/latest/api/app#event-session-created), [clientRequest.on('redirect')](https://www.electronjs.org/docs/latest/api/client-request#event-redirect).
* Some events are documented not to have an `Event`, but actually do, ex: [nativeTheme.on('updated')](https://www.electronjs.org/docs/latest/api/native-theme#event-updated), all the events in [powerMonitor](https://www.electronjs.org/docs/latest/api/power-monitor), many [webContents events](https://www.electronjs.org/docs/latest/api/web-contents#event-blur).
* Some events have a first argument that's called "event" but which doesn't inherit from Event, and doesn't have any Event-like properties or methods, e.g. [some Tray events](https://www.electronjs.org/docs/latest/api/tray#event-click), though [other Tray events](https://www.electronjs.org/docs/latest/api/tray#event-drop-text-macos) have a regular `Event` argument.

Reply:
Discussed with @MarshallOfSound offline and we're discussing removing the `sender` property from most events, as it's undocumented for everything except IPC events.